subsystems/intakesubsystem.py

```diff
@@ -0,0 +1,152 @@
+        Feeding = auto()  # arm in feeding position, motor ejecting
+        Staging = auto()  # arm is in position to score amp/trap, motor holding
+        Amp = auto()  # arm in same staging position, motor ejects
+        Trap = auto()  # arm pushes into trap, delay(?), motor ejects
+
+    def __init__(self) -> None:
+        Subsystem.__init__(self)
+        self.setName(__class__.__name__)  # basic subsystem boilerplate
+
+        self.pivotEncoder = CTREEncoder(
+            constants.kPivotEncoderID,
+            constants.kIntakeAngleOffset.degrees(),
+        )
+
+        self.intakeMotor = NEOBrushless(
+            constants.kIntakeCANID,
+            constants.kIntakeName,
+            constants.kIntakePIDSlot,
+            constants.kIntakePGain,
+            constants.kIntakeIGain,
+            constants.kIntakeDGain,
+            constants.kIntakeInverted,
+        )
+        self.pivotMotor = Talon(
+            constants.kPivotCANID,
+            constants.kPivotName,
+            constants.kPivotPGain,
+            constants.kPivotIGain,
+            constants.kPivotDGain,
+            constants.kPivotInverted,
+        )
+
+        # pivot motor spins 60 times per arm revolution
+        # use absolute encoder to determine motor position
+        pivotMotorPosition = (
+            self.pivotEncoder.getPosition().radians()
+            / constants.kRadiansPerRevolution
+            * constants.kPivotGearRatio
+        )
+        self.pivotMotor.setEncoderPosition(pivotMotorPosition)
+
+        self.state = self.IntakeState.Holding
+
+    def periodic(self) -> None:
+        # get actual velocity values for intake motor later
+
+        if self.state == self.IntakeState.Intaking:
+            self.pivotMotor.set(
+                Talon.ControlMode.Position,
+                constants.kFloorPositionAngle.radians()
+                / constants.kRadiansPerRevolution
+                * constants.kPivotGearRatio,
+            )
+            if self.intakeMotor.getLimitSwitch(NEOBrushless.LimitSwitch.Forwards):
+                self.intakeMotor.neutralOutput()
+            else:
+                self.intakeMotor.set(
+                    NEOBrushless.ControlMode.Velocity, constants.kIntakeSpeed
+                )
+
+        elif self.state == self.IntakeState.Holding:
+            self.pivotMotor.set(
+                Talon.ControlMode.Position,
+                constants.kHandoffAngle.radians()
+                / constants.kRadiansPerRevolution
+                * constants.kPivotGearRatio,
+            )
+            self.intakeMotor.neutralOutput()
+
+        elif self.state == self.IntakeState.Feeding:
+            self.pivotMotor.set(
+                Talon.ControlMode.Position,
+                constants.kHandoffAngle.radians()
+                / constants.kRadiansPerRevolution
+                * constants.kPivotGearRatio,
+            )
+            self.intakeMotor.set(
```

Comment:
feeding state still passes the note through in positive direction

do add an "eject" state which is intake in reverse (doubt we'd need to use it but still)